docs/hugepages.md

```diff
@@ -0,0 +1,40 @@
+# Backing Guest Memory by Huge Pages
+
+> \[!WARNING\]
+>
+> Support is currently in **developer preview**. See
+> [this section](RELEASE_POLICY.md#developer-preview-features) for more info.
+
+Firecracker supports backing the guest memory of a VM by 2MB hugetlbfs pages.
```

Comment:
Can we provide some guidance on when this may be beneficial?

Reply:
Sounds good, let me write something :)

Reply:
Done, please have another look

Reply:
looks good, thanks